src/figuresClasses.ts

```diff
@@ -1,21 +1,22 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
-  shape: 'triangle' | 'circle' | 'rectangle';
-  color: 'red' | 'green' | 'blue';
+  shape: Shape;
+  color: Color;
 
   getArea(): number;
 }
 
 export class Triangle implements Figure {
-  shape: 'triangle' | 'circle' | 'rectangle';
+  shape: Shape = 'triangle';
 
   constructor(
-    public color: 'red' | 'green' | 'blue',
+    public color: Color,
     public a: number,
     public b: number,
     public c: number,
   ) {
-    this.shape = 'triangle';
-
     if (a + b <= c || a + c <= b || b + c <= a) {
```

Comment:
you need to check in case one of the sides = 0
![image](https://github.com/mate-academy/ts_figures_classes/assets/112823560/7ab79501-552f-4708-ae0f-41d056d3935b)